src/cluster/cluster.go

```diff
@@ -0,0 +1,241 @@
+				return err
+			}
+			currURL := "http://" + c.SlaveDevices[i].ToFullAddress() + urlPath
+
+			resp, err := http.Post(currURL, "application/json", bytes.NewBuffer(body))
+			if err != nil {
+				log.Println("WARNING: Failed to POST to Peer: ", c.SlaveDevices[i].String(), currURL)
+				log.Println(err)
+			} else {
+				defer resp.Body.Close()
+				var result string
+				decoder := json.NewDecoder(resp.Body)
+				decoder.Decode(&result)
+				log.Println("Result:", result)
+			}
+		}
+	}
+	return nil
+}
+
+//******************************************************************************************************
+//*******Master Only Methods****************************************************************************
+//******************************************************************************************************
+
+//KingMe makes this device the master
+func (c *Cluster) KingMe() {
+	c.Me = device.Device{
+		ID:   c.generateUniqueID(),
+		Host: viper.GetString("GOFIRE_MASTER_HOST"),
+		Port: viper.GetString("GOFIRE_MASTER_PORT"),
+	}
+	c.MasterDevice = c.Me
+	//The Master waits ...
+}
+
+//AddDevice attempts to add a device to the cluster and returns the response data. This should only be run by the master.
+func (c *Cluster) AddDevice(newDevice device.Device) (response PeerUpdateMessage, err error) {
+	newDevice.ID = c.generateUniqueID()
+	c.SlaveDevices = append(c.SlaveDevices, newDevice)
+	c.PrintClusterInfo()
+
+	response = PeerUpdateMessage{
+		Source:  c.Me,
+		Cluster: *c,
+	}
+
+	exclusions := []device.Device{newDevice, c.Me}
+	err = c.UpdatePeers("/", response, exclusions)
+	if err != nil {
+		log.Println("Unexpected Error during attempt to contact all peers: ", err)
+		return PeerUpdateMessage{}, err
+	}
+
+	return response, nil
+}
+
+//******************************************************************************************************
+//*******Slave Only Methods*****************************************************************************
+//******************************************************************************************************
+
+//ALifeOfServitude is all that awaits this device
+func (c *Cluster) ALifeOfServitude() {
+	c.Me = device.Device{
+		ID:   c.generateUniqueID(),
+		Host: viper.GetString("GOFIRE_HOST"),
+		Port: viper.GetString("GOFIRE_PORT"),
+	}
+	masterHostname := viper.GetString("GOFIRE_MASTER_HOST") + ":" + viper.GetString("GOFIRE_MASTER_PORT")
+	//Try and Connect to the Master
+	err := test(masterHostname)
+	if err != nil {
+		log.Println("Failed to Reach Master Device: PANIC")
+		//TODO: Add Retry or failover maybe? panic for now
+		panic(err)
+	}
+	err = c.JoinNetwork(masterHostname)
+	if err != nil {
+		log.Println("Failed to Join Network: PANIC")
+		panic(err)
+	}
+}
+
+// JoinNetwork checks if the master exists and joins the network
+func (c *Cluster) JoinNetwork(URL string) error {
+	parsedURL, err := url.Parse("http://" + URL + "/join_network")
```

Comment:
This function is getting kind of long. Can we split it up a little? 

It looks like lines 127 - 136 creates-a-join-network-message.